devlab/tests/functional_test.py

```diff
@@ -149,8 +149,18 @@ def filter_vms(self):
         return [i for i in self.src_cloud.novaclient.servers.list(**opts)
                 if i.name in vms_names]
 
-    def filter_flavors(self):
-        flavors = [i['name'] for i in config.flavors]
+    def filter_flavors(self, filter_only_private=False):
+        flavors = []
+        all_flavors = config.flavors
+        for tenant in config.tenants:
+            if tenant.get('flavors'):
+                all_flavors += [flavor for flavor in tenant['flavors']]
+        for flavor in all_flavors:
+            if filter_only_private:
+                if 'is_public' in flavor and not flavor.get('is_public'):
```

Comment:
"if not flavor.get('is_public')" should be enough, "'is_public' in flavor" is not is not necessary here.